src/basic/cgroup-util.c

```diff
@@ -375,6 +409,12 @@ int cg_kill_recursive(
         assert(path);
         assert(sig >= 0);
 
+        if (sig == SIGKILL && FLAGS_SET(flags, CGROUP_REMOVE) && !FLAGS_SET(flags, CGROUP_IGNORE_SELF)) {
```

Comment:
i don't grok the conditionalization. If CGROUP_REMOVE is set we'll also try to remove the cgroup, but this codepaths now will not.

Also please add comment that we can ignore CGROUP_SIGCONT here, since it has no effect anyway if combined with SIGKILL.

Also, if `s` or log_kill is set, we should skip this shortcut, as in that case we need to record/log each process killed.

i.e. to me it appears the conditionalization should be at least.

```c
if (sig == SIGKILL && !FLAGS_SET(flags, CGROUP_IGNORE_SELF) && !s && !log_kill)
…
```

but this leaves the CGROUP_REMOVE handling unsolved, which needs to be addressed too. i.e. either skip the shortcut in that case, or pass execution on to cg_trim() or so.

Reply:
Writing to `cgroup.kill` automatically removes the cgroup, no? This appears to be the behavior on my machine, unless I'm misunderstanding the semantics of `CGROUP_REMOVE`.

Reply:
`cgroup.kill` is not meant to remove any cgroups that's a separate operation.

Reply:
as i understand things `cgroup.kill` doesn't imply "rmdir" as you suggest it does. @brauner?